### ISSUE TITLE
Wrong JSON behaviour for routes

### ISSUE THREAD

**Issue by leifg** [sportngin/okcomputer#53]
I tested this behaviour on Rails 4.2 and 4.1

According to the documentation a GET to /okcomupter.json will return a 200 OK with a JSON response. However on Rails 4.1 and 4.2 I will get a 404

    curl -I http://localhost:3001/okcomputer.json

    HTTP/1.1 404 Not Found
    Content-Type: text/html; charset=utf-8
    Content-Length: 12454
    X-Request-Id: 3f1c178b-008c-4a10-b706-a558dca5970b
    X-Runtime: 0.131676
    Server: WEBrick/1.3.1 (Ruby/2.2.0/2014-12-25)
    Date: Fri, 09 Jan 2015 11:49:07 GMT
    Connection: Keep-Alive

Request to /okcomputer/all.json is working though.

Furthermore I would expect setting the content type header to "application/json" would also reply with a json response but it doesn't.

    curl -H "Content-Type: application/json" http://localhost:3001/okcomputer/all
    default: PASSED Application is running
    database: PASSED Schema version: 0

tested it on an empty app with zero configuration (only gem "okcomputer" in Gemfile)

**Comment by pbyrne** [comment]
I'm able to reproduce the problem of Rails 4.2 apps not accepting `/okcomputer.json` (though `/okcomputer/default.json`, the equivalent request, *does* respond with JSON as expected). This seems to be some kind of routing difference in Rails 4.2 regarding the root route of engines. I haven't quite gotten to the bottom of it yet.

As to the second issue you report, with setting "Content-Type: application/json", note that the app looks at the HTTP Accept header to determine which response type to use, not the Content-Type header (which defines the content type of the request, not the expected content type of the response). This `curl` should get you what you want:

```
$ curl -H "Accept: application/json" http://localhost:3000/okcomputer/all    
{"default":{"message":"Application is running","success":true},"database":{"message":"Schema version: 0","success":true}}%                                                                                                  
```

**Comment by pbyrne** [comment]
I'm completely stumped on the 404 here. I've been able to confirm that *any* `HEAD` request (`curl -I` or `curl -XHEAD`) to *any* OK Computer route gets a 404 with a `ActionController::RoutingError (No route matches [HEAD] "/okcomputer/default.json")`. This feels like a bug in Rails 4.2, since `HEAD` requests should be treated identically to a `GET`, except without responding with the content.

Per [the Rails guide on Rack](http://guides.rubyonrails.org/rails_on_rack.html#internal-middleware-stack):

> Rack::Head
> 
> Converts HEAD requests to GET requests and serves them as so.

`HEAD` requests to other routes in our apps are properly routed; it's those requests to OK Computer that aren't getting the requests routed properly. I've confirmed identical behavior with Resque and MailView mounted as rack apps in our applications, where a `GET` succeeds but a `HEAD` gets a 404, so it seems to not be related to OK Computer specifically, but to Rails or Rack.
